src/main/java/uk/gov/hmcts/reform/managecase/security/SecurityUtils.java

```diff
@@ -56,8 +57,8 @@ private String removeBearerFromToken(String token) {
         return token.startsWith(BEARER) ? token.substring(BEARER.length()) : token;
     }
 
-    public boolean hasSolicitorRole(List<String> roles) {
-        return roles.stream().anyMatch(role -> SOLICITOR_ROLE.matcher(role).matches());
+    public boolean hasSolicitorRoleForJurisdiction(List<String> roles, String jurisdiction) {
+        return roles.stream().anyMatch(role -> SOLICITOR_ROLE.matcher(role).matches()
+            && role.toLowerCase(Locale.getDefault()).contains(jurisdiction.toLowerCase(Locale.getDefault())));
```

Comment:
I see we have reverted to a **contains** check rather than **starts-with**.  This still feels wrong to me as ccd-data store checks for a match on `"caseworker-" + jurisdictionId` as per [UserAuthorisation.hasJurisdictionRole](https://github.com/hmcts/ccd-data-store-api/blob/master/src/main/java/uk/gov/hmcts/ccd/infrastructure/user/UserAuthorisation.java#L25-L28).  

Is there a solicitor role out there that doesn't confirm to either: 
* `caseworker-jurisdiction-solicitor`
* `caseworker-jurisdiction-somethingelse-solicitor`

... that would justify using contains?  as matching on a simple short jurisdiction name (e.g. `IA`) looks a candidate to having false positives in the future.